istioctl/pkg/install/verify.go

```diff
@@ -31,41 +33,136 @@ import (
+	enableVerbose   bool
+}
+
+// NewVerifyCommand creates a new command for verifying Istio Installation Status
+func NewVerifyCommand() *cobra.Command {
+	vArgs := &verifyInstallArgs{}
+	verifyInstallCmd = &cobra.Command{
+		Use:   "verify-install",
+		Short: "Verifies Istio Installation Status or performs pre-check for the cluster before Istio installation",
+		Long: `
+		verify-install verifies Istio installation status against the installation file or the IstioControlPlane 
+		CustomResource you specified when you installed Istio. It loops through all the installation
+		resources defined in your installation file and reports whether all of them are
+		in ready status. It will report failure when any of them are not ready.
+
+		If you do not specify installation file it will perform pre-check for your cluster
+		and report whether the cluster is ready for Istio installation.
+`,
+		Example: `
+		# Verify that Istio can be freshly installed
+		istioctl verify-install
+		
+		# Verify the deployment matches a custom Istio deployment configuration
+		istioctl verify-install -f $HOME/istio.yaml
+
+		# Verify the deployment matches an IstioControlPlane CustomResource
+		istioctl verify-install -f $HOME/istio_v1alpha2_istiocontrolplane_cr.yaml
+
+		# Verify the deployment matches minimal profile for IstioControlPlane CustomResource
+		istioctl verify-install -s profile=minimal
+`,
+		RunE: func(c *cobra.Command, args []string) error {
+			l := mesh.NewLogger(false, c.OutOrStderr(), c.OutOrStderr())
+			return verifyInstall(vArgs, c.OutOrStderr(), args, l)
+		},
+	}
+
+	addVerifyInstallFlags(verifyInstallCmd, vArgs)
+	return verifyInstallCmd
+}
+
+func addVerifyInstallFlags(cmd *cobra.Command, v *verifyInstallArgs) {
+	var (
+		filenames     = []string{}
+		fileNameFlags = &genericclioptions.FileNameFlags{
+			Filenames: &filenames,
+			Recursive: boolPtr(false),
+			Usage:     "Istio YAML installation file or IstioControlPlane CustomResource file",
+		}
+		kubeConfigFlags = &genericclioptions.ConfigFlags{
+			Context:    strPtr(""),
+			Namespace:  strPtr(""),
+			KubeConfig: strPtr(""),
+		}
+	)
+	flags := cmd.PersistentFlags()
+	flags.StringVarP(&v.istioNamespace, "istioNamespace", "i", "istio-system",
+		"Istio system namespace")
+	verifyInstallCmd.PersistentFlags().StringSliceVarP(&v.set, "set", "s", nil, mesh.SetFlagHelpStr)
+	verifyInstallCmd.PersistentFlags().BoolVar(&v.enableVerbose, "enableVerbose", true,
+		"Enable verbose output")
+	kubeConfigFlags.AddFlags(flags)
+	v.kubeConfigFlags = kubeConfigFlags
+	fileNameFlags.AddFlags(flags)
+	v.fileNameFlags = fileNameFlags
+}
+
+func verifyInstall(v *verifyInstallArgs, writer io.Writer, args []string, l *mesh.Logger) error {
+	options := v.fileNameFlags.ToOptions()
+	if len(options.Filenames) == 0 && v.set == nil {
 		if len(args) != 0 {
 			_, _ = fmt.Fprint(writer, verifyInstallCmd.UsageString())
 			return fmt.Errorf("verify-install takes no arguments to perform installation pre-check")
```

Comment:
One of the error messages is `Istio cannot be installed because the Istio namespace '%v' is already in use`.  If the user supplies an option to set the namespace, or to use a IstioControlPlane with custom namespace, we should allow some args for pre-check.  Yes?

Reply:
Yes! Now I added --pre-check as an option and will respect the namespaces defined in icp or -set. 
Also will perform pre-check if no args and options provided.